js/download.js

```diff
@@ -116,3 +116,17 @@ function drawImageAndQuote(img, ctx) {
     );
   }
 }
+
+
+function saveToDevice(){
+  if (window.navigator.msSaveBlob) {
+    window.navigator.msSaveBlob(canvas.msToBlob(), "Wallpaper.jpg");
+  } else {
+    let a = document.createElement("a");
+    document.body.appendChild(a);
+    a.href = canvas.toDataURL('image/jpeg', 1.0);
+    a.download = "Wallpaper";
```

Comment:
It maybe better to ask the user for a name of his/her file

Or to use the first part of the quote with Image Id (Num) from  BG url ****.jpg in order if user uses the same quote

Or you can use `new Date().toDateString()` to generate a name

If you want to use time also you may think using something like this `new Date().toDateString()+' '+new Date().toLocaleTimeString().replaceAll(':','-')` because `:` prevented in filenames
`